### PR TITLE
draft

### PR DESCRIPTION
**Pull request by uarlouski**

### CODE REVIEW

File: vividus-facade-jira/src/main/java/org/vividus/jira/databind/JiraIssueDeserializer.java [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+
+        JsonNode fields = root.path("fields");
+        issue.setStatus(getFieldValue(fields, "status"));
+        issue.setResolution(getFieldValue(fields, "resolution"));
+        return issue;
+    }
+
+    private String getFieldValue(JsonNode root, String field)
+    {
+        return root.path(field).path("name").asText("undefined");
```
Comment (valfirst) [inline vividus-facade-jira/src/main/java/org/vividus/jira/databind/JiraIssueDeserializer.java:47]: why `"undefined"` ?

File: vividus-facade-jira/src/main/java/org/vividus/jira/model/JiraEntity.java [hunk 2]
```diff
@@ -1,5 +1,5 @@
 /*
- * Copyright 2019-2020 the original author or authors.
+ * Copyright 2019-2021 the original author or authors.
```
Comment (valfirst) [inline vividus-facade-jira/src/main/java/org/vividus/jira/model/JiraEntity.java:2]: noisy change

File: vividus-facade-jira/src/main/java/org/vividus/jira/model/Project.java [hunk 3]
```diff
@@ -1,5 +1,5 @@
 /*
- * Copyright 2019-2020 the original author or authors.
+ * Copyright 2019-2021 the original author or authors.
```
Comment (valfirst) [inline vividus-facade-jira/src/main/java/org/vividus/jira/model/Project.java:2]: noisy change

File: vividus-facade-jira/src/main/java/org/vividus/jira/model/Version.java [hunk 4]
```diff
@@ -1,5 +1,5 @@
 /*
- * Copyright 2019-2020 the original author or authors.
+ * Copyright 2019-2021 the original author or authors.
```
Comment (valfirst) [inline vividus-facade-jira/src/main/java/org/vividus/jira/model/Version.java:2]: noisy change

File: vividus-soft-assert/src/main/java/org/vividus/softassert/formatter/AssertionFormatter.java [hunk 5]
```diff
@@ -38,21 +43,16 @@ public String getMessage(String description, KnownIssue issue)
...
-        if (resolution != null)
-        {
-            message.append(DOT_CHAR).append(" Resolution: ").append(resolution);
-        }
-        message.append("). ").append(description);
-        return message.toString();
+                .append(" (");
+
+        Map<String, Object> details = new LinkedHashMap<>();
+        details.put("Type", issue.getType());
```
Comment (valfirst) [inline vividus-soft-assert/src/main/java/org/vividus/softassert/formatter/AssertionFormatter.java:49]: do you really need one more map? why not to add these values to string builder explicitly?

File: vividus-soft-assert/src/main/java/org/vividus/softassert/issue/IIssueStateProvider.java [hunk 6]
```diff
@@ -16,9 +16,31 @@
...
 package org.vividus.softassert.issue;
 
+import java.util.Map;
+
 public interface IIssueStateProvider
 {
-    String getIssueStatus(String issue);
+    IssueState getIssueState(String instance, String issue);
+
+    class IssueState
```
Comment (valfirst) [inline vividus-soft-assert/src/main/java/org/vividus/softassert/issue/IIssueStateProvider.java:25]: convert to external class

File: vividus-soft-assert/src/main/java/org/vividus/softassert/issue/IIssueStateProvider.java [hunk 7]
```diff
@@ -16,9 +16,31 @@
 
 package org.vividus.softassert.issue;
 
+import java.util.Map;
+
 public interface IIssueStateProvider
```
Comment (valfirst) [inline vividus-soft-assert/src/main/java/org/vividus/softassert/issue/IIssueStateProvider.java:21]:
```suggestion
public interface IssueStateProvider
```
?

File: vividus-soft-assert/src/main/java/org/vividus/softassert/issue/KnownIssueChecker.java [hunk 8]
```diff
@@ -22,21 +22,24 @@
...
 import java.util.regex.Pattern;
 
+import org.vividus.softassert.issue.IIssueStateProvider.IssueState;
 import org.vividus.softassert.model.KnownIssue;
 
 public class KnownIssueChecker implements IKnownIssueChecker
 {
     private final IKnownIssueProvider knownIssueProvider;
     private final KnownIssueDataProvider knownIssueDataProvider;
+    private final Optional<IIssueStateProvider> issueStateProvider;
```
Comment (valfirst) [inline vividus-soft-assert/src/main/java/org/vividus/softassert/issue/KnownIssueChecker.java:32]: List ?

File: vividus-soft-assert/src/main/java/org/vividus/softassert/model/KnownIssue.java [hunk 9]
```diff
@@ -26,20 +27,19 @@
 {
     private static final long serialVersionUID = 5419047883822146063L;
 
-    private static final String FIXED_ISSUE_STATUSES_STRING = "closed";
-    private static final String FIXED_ISSUE_RESOLUTIONS_STRING = "fixed|done";
-
     private final String identifier;
+    private final String bts;
```
Comment (valfirst) [inline vividus-soft-assert/src/main/java/org/vividus/softassert/model/KnownIssue.java:31]: Optional ?

File: vividus-soft-assert/src/main/java/org/vividus/softassert/model/KnownIssue.java [hunk 10]
```diff
@@ -26,20 +27,19 @@
...
+    private boolean fixed;
+    private Map<String, String> issueDetails;
 
     public KnownIssue(String identifier, KnownIssueIdentifier issueIdentifier, boolean potentiallyKnown)
     {
         this.identifier = identifier;
+        this.bts = issueIdentifier.getBts();
         this.type = issueIdentifier.getType();
         this.failTestCaseFast = issueIdentifier.isFailTestCaseFast();
         this.failTestSuiteFast = issueIdentifier.isFailTestSuiteFast();
```
Comment (valfirst) [inline vividus-soft-assert/src/main/java/org/vividus/softassert/model/KnownIssue.java:45]: just create a single field `issueIdentifier`?

**Review by valfirst — CHANGES_REQUESTED** [review]

File: vividus-connector-bts-jira/src/main/java/org/vividus/jira/connector/JiraConfiguration.java [hunk 11]
```diff
@@ -0,0 +1,80 @@
...
+
+import org.vividus.http.client.HttpClientConfig;
+
+public class JiraConfiguration
+{
+    private Pattern code;
+    private String endpoint;
+    private String username;
+    private String password;
+    private HttpClientConfig httpConfig = new HttpClientConfig();
```
[the remark this answers is not in the document]
Comment (uarlouski) [reply vividus-connector-bts-jira/src/main/java/org/vividus/jira/connector/JiraConfiguration.java:29]: Caused by: java.lang.NullPointerException
	at org.vividus.http.client.HttpClientFactory.buildHttpClient(HttpClientFactory.java:53) ~[main/:?]
checked

**Comment by uarlouski** [comment]
temp close

**Pull request closed without merging**